clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,375 @@
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void rename(TachyonURI src, TachyonURI dst)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Renames an existing Tachyon path to another Tachyon path in Tachyon. This operation will be
+   * reflected in the underlying storage if the path is persisted.
+   *
+   * @param src the path of the source, this must already exist
+   * @param dst the path of the destination, this path should not exist
+   * @param options options to associate with this operation
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given file does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void rename(TachyonURI src, TachyonURI dst, RenameOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #setAttribute(TachyonURI, SetAttributeOptions)} with default
+   * options.
+   *
+   * @param path the path to set attributes for
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given file does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void setAttribute(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Sets any number of a path's attributes.
+   *
+   * @param path the path to set attributes for
+   * @param options options to associate with this operation
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given file does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  void setAttribute(TachyonURI path, SetAttributeOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #unmount(TachyonURI, UnmountOptions)} with default options.
+   *
+   * @param path a Tachyon path, this must be a mount point
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
+   */
+  void unmount(TachyonURI path) throws IOException, TachyonException;
+
+  /**
+   * Unmounts a UFS subtree identified by the given Tachyon path. The Tachyon path match a
+   * previously mounted path. The contents of the subtree rooted at this path are removed from
+   * Tachyon but the corresponding UFS subtree is left untouched.
+   *
+   * @param path a Tachyon path, this must be a mount point
+   * @param options options to associate with this operation
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws TachyonException if a Tachyon exception occurs
+   */
+  void unmount(TachyonURI path, UnmountOptions options) throws IOException, TachyonException;
```

Comment:
If I do `unmount`, and then `mount` again, will the original path attributes be kept?